Author: Steffen Moeller <moeller@debian.org>
Description: Nothing in this patch should be functionally relevant, only compile time warnings as by the ambiguity of the result parameter are resolved.
--- a/clientgui/MainDocument.cpp
+++ b/clientgui/MainDocument.cpp
@@ -1640,20 +1640,20 @@
 }
 #else
 void CMainDocument::KillGraphicsApp(int pid) {
-    char* argv[6];
     char currentDir[1024];
-    char thePIDbuf[10];
+    char thePIDbuf[20];
     int id, iRetVal;
     
 
     if (g_use_sandbox) {
         snprintf(thePIDbuf, sizeof(thePIDbuf), "%d", pid);
-        argv[0] = "switcher";
-        argv[1] = "/bin/kill";
-        argv[2] =  "kill";
-        argv[3] = "-KILL";
-        argv[4] = thePIDbuf;
-        argv[5] = 0;
+        char* const argv[6] = {
+            strdup("switcher"),
+	    strdup("/bin/kill"),
+	    strdup("kill"),
+	    strdup("-KILL"),
+	    thePIDbuf,0
+        };
     
         iRetVal = run_program(
             getcwd(currentDir, sizeof(currentDir)),
@@ -1663,6 +1663,10 @@
             0,
             id
         );
+	free(argv[0]);
+	free(argv[1]);
+	free(argv[2]);
+	free(argv[3]);
     } else {
         kill_program(pid);
     }
